ts/setCompletedTodosDeletionListener.ts

```diff
@@ -0,0 +1,38 @@
+import { todos, todosUl } from "./todoapp.js";
+import { storage } from "./storage.js";
+import { Todo } from "./types.js";
+
+export const deleteCompletedButton = document.querySelector(
+  ".delete-completed"
+) as HTMLElement;
+
+export function setCompletedTodosDeletionListener() {
+  deleteCompletedButton.addEventListener("click", () => {
```

Comment:
This fn has O(n^2) complexity. It can be O(n)

When you click this button, you don't need to iterate over every UI element. Your localstorage represents todos, each todo has `completed flag`. 

You can take "all of completed" todos, iterate over them, at each iteration you find dom element and removes it. 

Then you need to update localstorage. Again, each task has `completed`. Just filter out completed tasks and that's your new localstorage value.